src/Tizen.Network.Stc/Tizen.Network.Stc/StcRule.cs

```diff
@@ -0,0 +1,352 @@
+            }
+        }
+
+        /// <summary>
+        /// Gets the statistics information an application matched rule asynchronously.
+        /// </summary>
+        /// <since_tizen> tizen_5.5 </since_tizen>
+        /// <param name="infoCb">The callback is called for each application that used network in between timestamps specified.</param>
+        /// <param name="userData">The user data passed to the callback function.</param>
+        /// <privilege>http://tizen.org/privilege/network.get</privilege>
+        /// <exception cref="NotSupportedException">Thrown when the Stc is not supported.</exception>
+        /// <exception cref="UnauthorizedAccessException">Thrown when the permission is denied.</exception>
+        /// <exception cref="InvalidOperationException">Thrown when the method failed due to an invalid operation.</exception>
+        public void GetStats(StatsInfoObjectCb.InfoCallback infoCb, IntPtr userData)
+        {
+            if (_disposed)
+            {
+                throw new ObjectDisposedException("Invalid StcRule instance (Object may have been disposed or released)");
+            }
+
+            _getStatsObjectCb = infoCb;
+
+            _getStatsCb = (int result, IntPtr info, IntPtr userDataBack) =>
+            {
+                if(result != (int)StcError.None)
+                {
+                    Log.Error(Globals.LogTag, "GetStats failed, Error - " + (StcError)result);
+                    StcErrorFactory.ThrowStcException(result);
+                }
+                return _getStatsObjectCb(new StcInfo(info), userDataBack);
+            };
+
+            int ret = Interop.Stc.Rule.GetStats(StcManagerImpl.Instance.GetSafeHandle(), _ruleHandle, _getStatsCb, userData);
+            if (ret != (int)StcError.None)
+            {
+                Log.Error(Globals.LogTag, "GetStats() failed , Error - " + (StcError)ret);
+                StcErrorFactory.ThrowStcException(ret);
+            }
+        }
+
+        /// <summary>
+        /// Gets the statistics information of each application asynchronously. The callback is called for each application that used network in between timestamps specified.
+        /// </summary>
+        /// <since_tizen> tizen_5.5 </since_tizen>
+        /// <param name="infoCb">The callback is called for each application that used network in between timestamps specified.</param>
+        /// <param name="userData">The user data passed to the callback function.</param>
+        /// <privilege>http://tizen.org/privilege/network.get</privilege>
+        /// <exception cref="NotSupportedException">Thrown when the Stc is not supported.</exception>
+        /// <exception cref="UnauthorizedAccessException">Thrown when the permission is denied.</exception>
+        /// <exception cref="InvalidOperationException">Thrown when the method failed due to an invalid operation.</exception>
+        public void ForeachStats(StatsInfoObjectCb.InfoCallback infoCb, IntPtr userData)
```

Comment:
```suggestion
        public IEnumerable<StcInfo> GetStats()
```

Reply:
The callback is called for each application to provide its information. So i can't  find what we have to await for inside this method. As we don't know how many times this callback will be called.

Reply:
See my suggestion. I recommend to use IEnumerable

Reply:
done